packages/ag-charts-website/src/astro/integrations/agCharts.ts

```diff
@@ -0,0 +1,61 @@
+import type { AstroIntegration } from 'astro';
+import { spawnSync } from 'child_process';
+import svgr from 'vite-plugin-svgr';
+
+import { getDevFileList } from '../../utils/pages';
+import agHotModuleReload from '../plugins/agHotModuleReload';
+
+export default function agCharts(): AstroIntegration {
+    return {
+        name: 'ag-charts',
+        hooks: {
+            'astro:config:setup': ({ updateConfig }) => {
+                // eslint-disable-next-line no-console
+                console.log('astro:config:setup');
+
+                updateConfig({
+                    vite: {
+                        plugins: [svgr(), agHotModuleReload()],
+                        resolve: {
+                            conditions: ['require'],
+                        },
+                        optimizeDeps: {
+                            exclude: ['ag-charts-community', 'ag-charts-enterprise'],
+                        },
+                        server: {
+                            fs: {
+                                allow: [
+                                    '.',
+                                    // Nx root node modules - for hmr (hot module reloading)
+                                    '../../node_modules/astro/dist/runtime/client',
+                                ].concat(getDevFileList()),
+                            },
+                        },
+                    },
+                });
+            },
+            'astro:build:done': () => {
+                // eslint-disable-next-line no-console
+                console.log('astro:build:done');
+
+                const start = performance.now();
+                spawnSync(
+                    'npx',
+                    [
+                        'prettier',
+                        './dist/packages/ag-charts-website/**/examples/**/*.{html,js,ts,jsx,tsx}',
+                        '--write',
+                        '--ignore-path',
+                        './dist/.prettierignore',
+                        '--log-level',
+                        'error',
+                    ],
+                    { stdio: 'inherit' }
+                );
```

Comment:
Rather than running prettier as a post-processing step on `dist/` outputs, could we instead use the JS API during generation, which may remove some of the overhead and other drawbacks to this approach?

Reply:
As discussed.

It is possible to do it like that for the js parts, but not for the html. Astro puts the example's html into an `.astro` template which then gets generated. There doesn't appear to be an integrated way in astro to run after that step other than in the `astro:build:done` hook.

It does mean the serve and build are not in parity, but that can be a future investigation.